### PR TITLE
refactor(efb/mcdu): move telex prompt out of MCDU and changes to PopUp class

### LINKED ISSUE #5726: EFB: Move the Telex Prompt and Setting from MCDU to EFB
### Aircraft Version

Development

### Description

Move the telex prompt (on first use) and setting from the MCDU to the EFB. It is one of the last 3 remaining "sim" options to be removed after #5344.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by ErickSharp**
Fixes #5726

## Summary of Changes

Removes the telex prompt out of MCDU and into the EFB's Settings > ATSU/AOC page. This also makes changes to the PopUp class made by @2hwk. Also removes RETURN selection on MCDU menu.

## Screenshots (if necessary)

Disabled: 
image

After Clicking on Toggle: 
image

Consent not yet given: 
image

MCDU Menu BEFORE:
image

MCDU Menu AFTER:
image

## Additional context

Discord username (if different from GitHub): Erick [Z-6]#6484

## Testing instructions

Go to the MCDU menu, there should NOT be an option on the right hand side that reads 'OPTIONS' Go to the EFB's ATSU/AOC settings page and confirm the Telex prompt is there. Finally, when going from disabled to on, make sure an MSFS dialogue pops up asking you to confirm your selection. 

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by ErickSharp** [comment]
I have a question to whoever reads this pull request, would you guys prefer a modal on the EFB or an in-sim dialog? I raise this question as I am not sure if it would prove to be too intrusive if you decide to toggle the option in-flight.

**Comment by Benjozork** [comment]
> 
> 
> I have a question to whoever reads this pull request: would you guys prefer a modal on the EFB or an in-sim dialog? I raise this question as I am not sure if it would prove to be too intrusive if you decide to toggle the option in-flight.

Sim dialog is fine

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js [hunk 1]
```diff
@@ -287,7 +287,7 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
         const onlineFeaturesStatus = NXDataStore.get("CONFIG_ONLINE_FEATURES_STATUS", "UNKNOWN");
 
         if (onlineFeaturesStatus === "UNKNOWN") {
-            CDU_OPTIONS_TELEX.ShowPage(this);
+            new NXNotif().showNotification({ message: "You have not yet configured the TELEX option. To do so, go to the electronic flight bag's settings page and set the TELEX setting on the ATSU/AOC tab. More information can be found at https://docs.flybywiresim.com/telex" });
```
Comment (Benjozork) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:290]: **thought:** Should we not show the dialog immediately like we did before ?

Comment (ErickSharp) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:290]: I thought about it but thought some might find it intrusive.

Comment (tracernz) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:290]: > 
> 
> I thought about it but thought some might find it intrusive.

That was intentional. 😛

Comment (ErickSharp) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:290]: > > I thought about it but thought some might find it intrusive.
> 
> That was intentional. 😛

Ok, sorry about that; I will make the change.

**Review by Benjozork — APPROVED** [review]

**Comment by ShadowFrost3** [comment]
Quality Assurance Trainee Report


Object of testing: #5902
Tier of Testing:1
Date: 26/09/2021

Testing Process: 
C&D
Look at MCDU, click on EFB, do the things requested - checked

Negatives: 
Testing Results: Passed

Conclusions: 
image
image
image
image

**Comment by tracernz** [comment]
Part of the testing should include checking what happens when A32NX is first installed (telex setting never set before). In that case there is a pop-up. It can be simulated by running `NXDataStore.set("CONFIG_ONLINE_FEATURES_STATUS", "UNKNOWN")` in the debugger, and restarting the flight.

**Comment by Samueleonard** [comment]
Quality Assurance Tester Report

Discord: 1samueleonard1#0001
Object of testing: (#5902 )
Tier of Testing: (1)
Date: (27/09/2021)

Testing Process: 
. looked in mcdu - option menu is now gone
. looked in efb - telex option is now there
. enabled/disabled option - popup appeared - tested with different options
Negatives: n/a
Testing Results: (Passed)

Conclusions: lgtm, my OCD is now happy with no option page in the mcdu
. did not actually test the telex feature, but i see no reason why it would be broke
. did not test a completely clean install
Media: n/a

**Review by tracernz — CHANGES_REQUESTED** [review]
Telex definitely works when it's enabled. The new user popup buttons don't work though due to the issue noted.

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js [hunk 2]
```diff
@@ -287,7 +287,13 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
         const onlineFeaturesStatus = NXDataStore.get("CONFIG_ONLINE_FEATURES_STATUS", "UNKNOWN");
 
         if (onlineFeaturesStatus === "UNKNOWN") {
-            CDU_OPTIONS_TELEX.ShowPage(this);
+            new NXPopUp().showPopUp(
+                'TELEX CONFIGURATION',
+                'You have not yet configured the telex option. Telex enables free text and live map. If enabled, aircraft position data is published for the duration of the flight. Messages are public and not moderated. USE AT YOUR OWN RISK. To learn more about telex and the features it enables, please go to https://docs.flybywiresim.com/telex. Would you like to enable telex?',
+                'small',
+                () => setTelexEnabled('ENABLED'),
+                () => setTelexEnabled('DISABLED'),
```
[the remark this answers is not in the document]
Comment (ErickSharp) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:295]: Ah yeah, my bad, I forgot that I just copied and pasted React code from before over.

**Review by tracernz — APPROVED** [review]

**Comment by GhostEagle68** [comment]
Quality Assurance Tester/Trainee Report

Discord: GhostEagle68#0030
Object of testing: #5902
Tier of Testing: 1
Date: 09/28/2021

Testing Process: 

1. -Spawned at PANC Gate C3
2. -Turn on ground power
3. -Turned on BATT 1+2
4. -Went into MCDU Menu
   a. Confirmed no OPTIONS menu
5. - Checked EFB's ATSU/AOC Setting Page
   a. Confirmed TELEX Option was there
   b. Confirmed message when toggling TELEX on
6. Tested rapidly turning off and on TELEX, worked as normal. No CTD detected

Negatives:       
N/A

Testing Results: 
Passed!

Conclusions: 
-Selecting NO on message leaves TELEX off
-Message appears every time it's toggled on and off

Media:       
OPTIONS not in MCDU Menu 
image

TELEX Option Visible
image

TELEX Message when switching option on
image

**Comment by tracernz** [comment]
I just tested again the new user experience... and it's good now. Nice work. 👍

**Pull request merged**